drivers/usb/core/usb.c

```diff
@@ -273,9 +273,13 @@ EXPORT_SYMBOL_GPL(usb_find_alt_setting);
 struct usb_interface *usb_ifnum_to_if(const struct usb_device *dev,
 				      unsigned ifnum)
 {
-	struct usb_host_config *config = dev->actconfig;
+	struct usb_host_config *config = NULL;
 	int i;
 
+	if (!dev)
+		return NULL;
```

Comment:
I'm really no linux hacker, but wouldn't it make sense to move this check all the way up to the opening curly brace? Initializing the struct and integer is of no use, if the device is NULL.

Reply:
The code is fine. You need all declarations to precede code in the C standard kernel targets.

Reply:
I wanted to do that too but C best practices - declare variables first.  Some compilers may have issue with the order.